modules/dnn/src/layers/layers_common.simd.hpp

```diff
@@ -820,15 +820,17 @@ void fastGEMM1T( const float* vec, const float* weights,
         for( int k = 0; k < vecsize; k += 8, wptr += 8 )
         {
             vfloat32m2_t v = vle32_v_f32m2(vec + k, vl);
-
-            vs0 = vfmacc_vv_f32m2(vs0, vle32_v_f32m2(wptr, vl), v, vl);
-            vs1 = vfmacc_vv_f32m2(vs1, vle32_v_f32m2(wptr + wstep, vl), v, vl);
-            vs2 = vfmacc_vv_f32m2(vs2, vle32_v_f32m2(wptr + wstep*2, vl), v, vl);
-            vs3 = vfmacc_vv_f32m2(vs3, vle32_v_f32m2(wptr + wstep*3, vl), v, vl);
-            vs4 = vfmacc_vv_f32m2(vs4, vle32_v_f32m2(wptr + wstep*4, vl), v, vl);
-            vs5 = vfmacc_vv_f32m2(vs5, vle32_v_f32m2(wptr + wstep*5, vl), v, vl);
-            vs6 = vfmacc_vv_f32m2(vs6, vle32_v_f32m2(wptr + wstep*6, vl), v, vl);
-            vs7 = vfmacc_vv_f32m2(vs7, vle32_v_f32m2(wptr + wstep*7, vl), v, vl);
+            switch (mvl) {
+                case 8: vs7 = vfmacc_vv_f32m2(vs7, vle32_v_f32m2(wptr + wstep*7, vl), v, vl); // fall through
+                case 7: vs6 = vfmacc_vv_f32m2(vs6, vle32_v_f32m2(wptr + wstep*6, vl), v, vl); // fall through
+                case 6: vs5 = vfmacc_vv_f32m2(vs5, vle32_v_f32m2(wptr + wstep*5, vl), v, vl); // fall through
+                case 5: vs4 = vfmacc_vv_f32m2(vs4, vle32_v_f32m2(wptr + wstep*4, vl), v, vl); // fall through
+                case 4: vs3 = vfmacc_vv_f32m2(vs3, vle32_v_f32m2(wptr + wstep*3, vl), v, vl); // fall through
+                case 3: vs2 = vfmacc_vv_f32m2(vs2, vle32_v_f32m2(wptr + wstep*2, vl), v, vl); // fall through
+                case 2: vs1 = vfmacc_vv_f32m2(vs1, vle32_v_f32m2(wptr + wstep, vl), v, vl); // fall through
+                case 1: vs0 = vfmacc_vv_f32m2(vs0, vle32_v_f32m2(wptr, vl), v, vl); break;
+                default: break;
+            }
```

Comment:
Switch at every loop iteration is bad idea. I propose to do the following:
- Reduce loop iterations by one, if vecsize is not dividable by 8.
- Run remaining iterations with step1 using RVV.
Loop unrolling by 8 is helps to perform part of computation then.

Reply:
I agree. And there are two tail part in fastGEMM: "vecsize" and "nvecs". 

I have rewritten the code by loop unrolling and taking minimum to prevent out-of-bounds memory access.